pkg/logs/processor/processor.go

```diff
@@ -97,12 +173,19 @@ func (p *Processor) run() {
 		p.done <- struct{}{}
 	}()
 
-	for msg := range p.inputChan {
-		// process the message
-		p.processMessage(msg)
-		p.mu.Lock() // block here if we're trying to flush synchronously
-		//nolint:staticcheck
-		p.mu.Unlock()
+	for {
+		select {
+		case msg, ok := <-p.inputChan:
+			if !ok {
+				return
+			}
```

Comment:
Any reason to add this validation compared to the original code?
I don't see a problem with it I'm just wondering if I've missed something.

Reply:
Got it, indeed!